src/test/java/com/synopsys/integration/jenkins/detect/service/DetectArgumentServiceTest.java

```diff
@@ -0,0 +1,203 @@
+package com.synopsys.integration.jenkins.detect.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import com.synopsys.integration.jenkins.extensions.JenkinsIntLogger;
+import com.synopsys.integration.jenkins.wrapper.JenkinsVersionHelper;
+import com.synopsys.integration.log.LogLevel;
+import com.synopsys.integration.util.IntEnvironmentVariables;
+
+import hudson.model.TaskListener;
+
+public class DetectArgumentServiceTest {
+    private final JenkinsVersionHelper jenkinsVersionHelper = Mockito.mock(JenkinsVersionHelper.class);
+
+    private static final String errorMessage = "Output Detect Command does not contain: ";
+    private static final String loggingLevelKey = "--logging.level.com.synopsys.integration";
+    private static final String expectedTestInvocationParameter = "TestInvocationParameter";
+    private static final String expectedJenkinsVersion = "JenkinsVersion";
+    private static final String pluginName = "blackduck-detect";
+    private static final String expectedJenkinsPluginVersion = "JenkinsPluginVersion";
+    private static final String jenkinsVersionParam = "--detect.phone.home.passthrough.jenkins.version";
+    private static final String pluginVersionParam = "--detect.phone.home.passthrough.jenkins.plugin.version";
+
+    private final IntEnvironmentVariables intEnvironmentVariables = new IntEnvironmentVariables();
+    private final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
```

Comment:
Since we're initializing the maps in the setUp method because we don't want to rely on JUnit cleaning them up for us, it makes sense to also do the same with the OutputStream and the EnvironmentVariables.